### PR TITLE
:running: Add script to run tests in Prow

### LINKED ISSUE #276: Run tests via prow
We should be running our test suite in prow instead of Travis, so the normal kube workflow commands work.

### PR DESCRIPTION
**Pull request by alvaroaleman**
This PR adds a script to run tests in Prow. Done to resolve #276

### CODE REVIEW

**Comment by alvaroaleman** [comment]
Requires https://github.com/kubernetes/test-infra/pull/14162 to merge, so we can verify this works. I tested this on a different Prow instance so I am reasonably confident there wont be issues.

Should I already remove the `.travis.yaml` or do we want to merge this -> Make Presubmit mandatory -> Remove `.travis.yaml`?

/assign @DirectXMan12

**Comment by DirectXMan12** [comment]
yeah, let's get mandatory presubmit merged before we remove travis.  Better to run tests twice than not at all.

File: hack/ci-check-everything.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+export TRACE=1
+
+# Not included by default in Prow
+export PATH=$(go env GOPATH)/bin:$PATH
+
+echo "Installing dep"
+(
+	# Doesn't work with gomodules turned on, see https://github.com/golang/dep/issues/2055
+	export GO111MODULE=off
+	go get -u github.com/golang/dep/cmd/dep
```
Comment (DirectXMan12) [inline hack/ci-check-everything.sh]: should we just be downloading the compiled binary here?  Prob *way* quicker.

**Comment by alvaroaleman** [comment]
@DirectXMan12 updated the script to install a binary version of `dep` instead of using `go get`.

Could you also have a look on the pull in test-infra? It adds the new presubmit as optional, so it is safe to merge already. Once it is merged, we can verify here if everything works :)

**Comment by DirectXMan12** [comment]
yep, looked at it.  Looks like @droot isn't in the kubernetes org, but otherwise LGTM

**Comment by DirectXMan12** [comment]
hold off on approving this till that merges?

**Comment by alvaroaleman** [comment]
Okay, so the presubmit passed, with a timeout on the first attempt. My best guess as why that happens is CPU starvation, which is why I opened https://github.com/kubernetes/test-infra/pull/14184 to make the job request CPU.

IMHO this is good to go, PTAL

**Comment by DirectXMan12** [comment]
/lgtm
/approve

awesome

excited for this

**Pull request merged**